internal/types/tx_indexer.go

```diff
@@ -0,0 +1,20 @@
+package types
+
+type TxStatus string
+type TxOnChainStatus string
+
+const (
+	TxProposed       TxStatus        = "PROPOSED"
+	TxVerified       TxStatus        = "VERIFIED"
+	TxSigned         TxStatus        = "SIGNED"
+	TxOnChainPending TxOnChainStatus = "PENDING"
+	TxOnChainSuccess TxOnChainStatus = "SUCCESS"
+	TxOnChainFail    TxOnChainStatus = "FAIL"
+)
+
+type Chain string
```

Comment:
Let's reuse `common.Chain` so we only ahve one way to reference chains throughout the codebase